Makefile

```diff
@@ -61,7 +61,15 @@ $(DOCKER_DISTRIB):
 	curl -o $(DOCKER_DISTRIB) $(DOCKER_DISTRIB_URL)
 
 tests:
-	go test -cover -tags netgo ./...
+	echo "mode: count" > profile.cov
+	for dir in $$(find . -type f -name '*_test.go' | xargs -n1 dirname | sort | uniq); do \
+	    go test -tags netgo -covermode=count -coverprofile=$$dir/profile.tmp $$dir;       \
+	    if [ -f $$dir/profile.tmp ]; then                                                 \
+	        cat $$dir/profile.tmp | tail -n +2 >> profile.cov;                            \
```

Comment:
No need to invoke cat there:

    tail -n +2 <$$dir/profile.tmp >> profile.cov

Also, does the temp file need to be in the directory?  Better to put it under tmp if it doesn't (e.g. `${TMPDIR:-/tmp}/profile.$$`).